CMakeLists.txt

```diff
@@ -1,30 +1,43 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(lms1xx)
 
+add_compile_options(-std=c++11)
+
 # Build ROS-independent library.
 find_package(console_bridge REQUIRED)
-include_directories(include ${console_bridge_INCLUDE_DIRS})
+include_directories(include SYSTEM ${console_bridge_INCLUDE_DIRS})
+
+# CoLaA Library abstracting protocol and implementing LMS1xx communication
+add_library(CoLaA src/colaa.cpp src/parse_helpers.cpp)
+target_link_libraries(CoLaA ${console_bridge_LIBRARIES})
 
-add_library(LMS1xx src/LMS1xx.cpp)
-target_link_libraries(LMS1xx ${console_bridge_LIBRARIES})
+# Specialisations for LMS5xx series scanners
+add_library(LMS5xx src/lms5xx.cpp)
+target_link_libraries(LMS5xx CoLaA ${console_bridge_LIBRARIES})
 
 add_library(MRS1000 src/MRS1000.cpp)
-target_link_libraries(MRS1000 LMS1xx ${console_bridge_LIBRARIES})
+target_link_libraries(MRS1000 CoLaA ${console_bridge_LIBRARIES})
 
 
 # Regular catkin package follows.
 find_package(catkin REQUIRED COMPONENTS roscpp sensor_msgs)
-catkin_package(CATKIN_DEPENDS roscpp)
+catkin_package(CATKIN_DEPENDS roscpp sensor_msgs)
+
+include_directories(include SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
You will still get the warnings for those libraries when you build them by themselves (which is where all warnings should be fixed imo), just not when you build this node.